### PR TITLE
[WIP] Micro endpoints

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Added the concept of micro endpoint which is a minimal NSB thing that only has a receiver and a single-stage pipeline.

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    public class When_sending_to_a_node: NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_receive_the_message()
+        {
+            var nodeConfig = new MicroEndpointConfiguration(new LogicalAddress(new EndpointInstance("SendingToANode.Receiver", null, null)));
+            nodeConfig.UseTransport<MsmqTransport>();
+            var processor = new NodeProcessingBehavior();
+            nodeConfig.Pipeline.Register("Processor", typeof(NodeProcessingBehavior), "Processor");
```
Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs]: @SzymonPobiega I plan to add an overload where you can add an instance of a behavior. Will make this easier

File: src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    using NServiceBus.Routing;
+    using NServiceBus.Transports;
+    using NUnit.Framework;
+
+    public class When_sending_to_a_node: NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_receive_the_message()
+        {
+            var nodeConfig = new MicroEndpointConfiguration(new LogicalAddress(new EndpointInstance("SendingToANode.Receiver", null, null)));
```
Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs]: Don't like passing an endpoint instance. Some kind of "receive address" would make more sense

File: src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs (hunk 1 above)
Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs]: @danielmarbach already did in on #3118

File: src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs (hunk 2 above)
Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/Node/When_sending_to_a_node.cs]: I think logical address makes sense here. We are transport-independent. The question is weather it should be mandatory to pass endpoint instance name in order to create a logical address.

**Comment by andreasohlund** [comment]
@SzymonPobiega what's the next steps? Close this pull and raise an issue to track the idea?

**Comment by SzymonPobiega** [comment]
Intentionally I did not delete the branch (yet).

**Pull request closed without merging**